test/controllers/api_users/permissions_controller_test.rb

```diff
@@ -343,5 +368,32 @@ class ApiUsers::PermissionsControllerTest < ActionController::TestCase
 
       patch :update, params: { api_user_id: api_user, application_id: application, application: { supported_permission_ids: [permission.id] } }
     end
+
+    should "exclude applications with revoked access tokens" do
+      sign_in create(:superadmin_user)
+
+      application = create(:application)
+      api_user = create(:api_user)
+      create(:access_token, resource_owner_id: api_user.id, application:, revoked_at: Time.zone.now)
+
+      assert_raises(ActiveRecord::RecordNotFound) do
+        patch :update, params: { api_user_id: api_user, application_id: application, application: { supported_permission_ids: [] } }
+      end
+    end
+
+    should "include applications with revoked access tokens when there is at least one non-revoked access token" do
+      sign_in create(:superadmin_user)
+
+      application = create(:application, with_supported_permissions: %w[permission])
+      api_user = create(:api_user)
+      create(:access_token, resource_owner_id: api_user.id, application:, revoked_at: Time.zone.now)
```

Comment:
Minor: You can use `Time.current` instead of `Time.zone.now` - I thought rubocop enforced that, but I must've imagined it!

Reply:
A quick search in the project suggests that we're using `Time.zone.now` in more places than `Time.current` so I'm not going to change this for now. Thanks for the heads up, though :+1: